From 38bbb6cf8be2f51a361dfc792af9934792069725 Mon Sep 17 00:00:00 2001
From: "Yang, Dong" <dong.yang@intel.com>
Date: Wed, 29 Apr 2015 15:10:09 +0800
Subject: [PATCH 408/462] adjust sensor registers to reduce reddish during
 sensor opening

the camera display reddish during opening, this is gc2235 sensor
issue, and it can not be 100% removed as confirmed by the vendor.
adjust the sensor register to reduce the reddish as much as possible.

Change-Id: I3dfdb9bcca2f2467f2cebdc8b8c7ad3b92035b17
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-31108
Signed-off-by: Yang, Dong <dong.yang@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/54280
Validation-Android: Wei Tao <wei.tao@intel.com>
Build: IMIN (SC)
Reviewed-by: Shuguang Gong <shuguang.gong@intel.com>
Approver: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Approver: Yunliang Ding <yunliang.ding@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Build: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Maintainer: WeiX Meng <weix.meng@intel.com>
---
 .../camera/drivers/media/i2c/gc2235.h              | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/gc2235.h b/drivers/external_drivers/camera/drivers/media/i2c/gc2235.h
index ad80db8..eb5b380 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/gc2235.h
+++ b/drivers/external_drivers/camera/drivers/media/i2c/gc2235.h
@@ -253,7 +253,7 @@ static struct gc2235_reg const gc2235_init_settings[] = {
 	{ GC2235_8BIT, 0x0f, 0x06 }, /* win width: 1616 */
 	{ GC2235_8BIT, 0x10, 0x60 },
 	{ GC2235_8BIT, 0x17, 0x15 }, /* mirror flip */
-	{ GC2235_8BIT, 0x18, 0x12 },
+	{ GC2235_8BIT, 0x18, 0x1a },
 	{ GC2235_8BIT, 0x19, 0x06 },
 	{ GC2235_8BIT, 0x1a, 0x01 },
 	{ GC2235_8BIT, 0x1b, 0x4d },
@@ -268,17 +268,17 @@ static struct gc2235_reg const gc2235_init_settings[] = {
 	{ GC2235_8BIT, 0x27, 0x30 },
 	{ GC2235_8BIT, 0x3f, 0x00 }, /* PRC */
 	/* blk */
-	{ GC2235_8BIT, 0x40, 0x03 },
-	{ GC2235_8BIT, 0x41, 0x00 },
+	{ GC2235_8BIT, 0x40, 0xa3 },
+	{ GC2235_8BIT, 0x41, 0x82 },
 	{ GC2235_8BIT, 0x43, 0x20 },
-	{ GC2235_8BIT, 0x5e, 0x00 },
-	{ GC2235_8BIT, 0x5f, 0x00 },
-	{ GC2235_8BIT, 0x60, 0x00 },
-	{ GC2235_8BIT, 0x61, 0x00 },
-	{ GC2235_8BIT, 0x62, 0x00 },
-	{ GC2235_8BIT, 0x63, 0x00 },
-	{ GC2235_8BIT, 0x64, 0x00 },
-	{ GC2235_8BIT, 0x65, 0x00 },
+	{ GC2235_8BIT, 0x5e, 0x18 },
+	{ GC2235_8BIT, 0x5f, 0x18 },
+	{ GC2235_8BIT, 0x60, 0x18 },
+	{ GC2235_8BIT, 0x61, 0x18 },
+	{ GC2235_8BIT, 0x62, 0x18 },
+	{ GC2235_8BIT, 0x63, 0x18 },
+	{ GC2235_8BIT, 0x64, 0x18 },
+	{ GC2235_8BIT, 0x65, 0x18 },
 	{ GC2235_8BIT, 0x66, 0x20 },
 	{ GC2235_8BIT, 0x67, 0x20 },
 	{ GC2235_8BIT, 0x68, 0x20 },
-- 
1.9.1

